File: 1803-average-waiting-time/average-waiting-time.py
class Solution:
    def averageWaitingTime(self, cus: List[List[int]]) -> float:
        n=len(cus)
        tot=0;cur=0
        for i in range(n):
            if cur<cus[i][0]:
                cur=cus[i][0]+cus[i][1]
                tot+=cus[i][1]
            else:
                cur+=cus[i][1]
                tot+=(cur-cus[i][0])

        return tot/n

File: 1803-average-waiting-time/average-waiting-time.cs
public class Solution {
    public double AverageWaitingTime(int[][] cus) {
        int n=cus.Length;
        long tot=0,cur=0;
        for(int i=0;i<n;i++){
            if(cur<cus[i][0]){
                cur=cus[i][0]+cus[i][1];
                tot+=cus[i][1];
            }
            else{
                cur+=cus[i][1];
                tot+=(cur-cus[i][0]);
            }
        }
        double ans=(double)tot/(double)n;
        return ans; 
    }
}